snuba/utils/streams/synchronized.py

```diff
@@ -0,0 +1,203 @@
+from dataclasses import dataclass
+from threading import Event
+from typing import Callable, Mapping, MutableMapping, Optional, Sequence, Set
+
+from snuba.utils.concurrent import Synchronized, execute
+from snuba.utils.streams.consumer import Consumer, ConsumerError, EndOfPartition
+from snuba.utils.streams.types import Message, Partition, Topic, TPayload
+
+
+@dataclass(frozen=True)
+class Commit:
+    __slots__ = ["group", "partition", "offset"]
+
+    group: str
+    partition: Partition
+    offset: int
+
+
+class SynchronizedConsumer(Consumer[TPayload]):
```

Comment:
yes, this is definitively simpler than the other SynchronizedConsumer we have :)